compiler-rt/lib/sanitizer_common/sanitizer_linux.cpp

```diff
@@ -1287,7 +1287,7 @@ uptr GetPageSize() {
 
 uptr ReadBinaryName(/*out*/ char *buf, uptr buf_len) {
 #  if SANITIZER_HAIKU
-  int32_t cookie = 0;
+  int32 cookie = 0;
   image_info info;
   const char *argv0 = "<UNKNOWN>";
   while (get_next_image_info(B_CURRENT_TEAM, &cookie, &info) == B_OK) {
```

Comment:
could be a follow-up PR if desired. there is a simpler way to get this info.

https://www.haiku-os.org/docs/api/group__libroot.html#gad7aea8c7f59a75196467df38f9c5635a

e.g. like [here](https://github.com/yquake2/yquake2remaster/blob/806db2d95251706d0e74e6abf87fb1eea59354d8/src/backends/generic/misc.c#L132).